drivers/disk/sdmmc_subsys.c

```diff
@@ -93,12 +93,9 @@ static int disk_sdmmc_access_ioctl(struct disk_info *disk, uint8_t cmd, void *bu
 	case DISK_IOCTL_CTRL_INIT:
 		return disk_sdmmc_access_init(disk);
 	case DISK_IOCTL_CTRL_DEINIT:
-		sdmmc_ioctl(&data->card, DISK_IOCTL_CTRL_SYNC, NULL);
-		/* sd_init() will toggle power to SDMMC, so we can just mark
-		 * disk as uninitialized
-		 */
+		/* Card will be uninitialized after DEINIT */
```

Comment:
We still need to sync the disk here, right? Otherwise removing power like this isn't a safe operation

Reply:
The Sync operation has been added to the deinit ioctl in the disk ops file.

Reply:
Sorry, I see that now- missed it since you are just using the SD functions directly.